src/mscorlib/shared/System/Runtime/InteropServices/MemoryMarshal.Fast.cs

```diff
@@ -39,6 +39,20 @@ public static partial class MemoryMarshal
         /// </summary>
         public static ref T GetReference<T>(ReadOnlySpan<T> span) => ref span._pointer.Value;
 
+        /// <summary>
+        /// Returns a reference to the 0th element of the Span. If the Span is empty, returns a reference to fake non-null pointer. Such a reference can be used
+        /// for pinning but must never be dereferenced. This is useful for interop with methods that do not accept null pointers for zero-sized buffers.
+        /// </summary>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        internal static unsafe ref T GetNonNullPinnableReference<T>(Span<T> span) => ref (span.Length != 0) ? ref span._pointer.Value : ref Unsafe.AsRef<T>((void*)1);
```

Comment:
Do we need both Read/Write and ReadOnly overloads of this?

Reply:
This does not matter a whole lot because of this is internal method. We can have discussion about this if we decide to make these public.

- It mirrors GetReference that has both even though both are not strictly needed (but it is probably a good idea to save JIT some work on optimizing out the casts).
- If we did not have both overloads, some of the callsites would need to have explicitly specified generic instantiation or explicit cast.